app/scripts/controllers/MainCtrl.js

```diff
@@ -1,6 +1,7 @@
 (function () {
 	function MainCtrl($scope, Room) {
 		$scope.rooms = Room.all;
+		this.addRoom = Room.addRoom;
```

Comment:
Why `this` when we have `$scope`? Add `console.log(this, $scope);` to the `MainCtrl` function and check the output in the browser.

Also, setting a property to a method isn't a great practice, especially since this line doesn't convey what's actually happening. You are executing `Room.addRoom` in the view rather than letting the controller be the intermediary, and here I would never know there's an expected argument to `addRoom`. A better solution is to make `$scope.addRoom(newName)` a controller function that in turn calls `Room.addRoom(newName)`.

Once you've created that function and called `Room.addRoom` you'll find this function is the perfect place to use `$scope.roomName = null;` to empty out the input.